### PR TITLE
Add timezone set/get and datetime function

### LINKED ISSUE #1289: Support date and time functions

### PR DESCRIPTION
**Pull request by laura-ding**
close #1289

- Add interface to set system timezone and get system timezone
- Add time function
- Modify timestamp display

### CODE REVIEW

File: src/common/filter/FunctionManager.cpp [hunk 1]
```diff
@@ -272,7 +274,7 @@ FunctionManager::FunctionManager() {
         auto &attr = functions_["length"];
         attr.minArity_ = 1;
         attr.maxArity_ = 1;
-        attr.body_ = [] (const auto &args) {
+        attr.body_ = [] (const auto &args, const auto*) {
```
Comment (darionyaphet) [inline src/common/filter/FunctionManager.cpp]: why change to pointer？Does the reference have some problem ?

File: src/common/filter/TimeFunction.h [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+#include "Expressions.h"
+
+/**
+ * Time functions can calculate from 1400 years, and the ones before 1400 years are not supported
+ */
+
+namespace nebula {
+class TimeCommon final {
+public:
+    TimeCommon() = default;
```
Comment (darionyaphet) [inline src/common/filter/TimeFunction.h]: maybe `TimeCommon` don't need to create instance? so we could use delete ?

File: src/common/filter/test/CMakeLists.txt [hunk 3]
```diff
@@ -26,6 +26,14 @@ nebula_add_test(
     LIBRARIES gtest gtest_main ${THRIFT_LIBRARIES} wangle
 )
 
+nebula_add_test(
+    NAME timefunction_test
```
Comment (darionyaphet) [inline src/common/filter/test/CMakeLists.txt]: time_function_test

File: src/common/filter/FunctionManager.cpp (hunk 1 above)
Comment (laura-ding) [reply src/common/filter/FunctionManager.cpp]: It is timezone，mostly function doesn't need it.

File: src/common/filter/TimeType.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    ErrorType = -1,
+    NullType = 0,
+    DataType = 1,
+    DataTimeType = 2,
+    TimeType = 3,
+    TimestampType = 4,
+};
+
+struct Interval {
+    int16_t  year;
```
Comment (darionyaphet) [inline src/common/filter/TimeType.h]: the type should be uint

**Pull request closed without merging**